velox/type/DecimalUtil.h

```diff
@@ -59,7 +62,7 @@ class DecimalUtil {
     }
     // Check overflow.
     if (rescaledValue < -DecimalUtil::kPowersOfTen[toPrecision] ||
-        rescaledValue > DecimalUtil::kPowersOfTen[toPrecision]) {
```

Comment:
You're test is exposing an issue here, toPrecision is 38 is causing is to read off the end of the array DecimalUtil::kPowersOfTen which has length 38.

Shouldn't kPowersOfTen go to 10^38 if we support precision up to and including 38?

Otherwise we need to check if scaleDifference  and toPrecision are < 38 before attempting to access these arrays.

Reply:
@kevinwilfong you are right. I have extended kPowersOfTen in this PR: https://github.com/facebookincubator/velox/pull/2456

Reply:
I will be moving the change of extending kPowersOfTen to 38 in this PR and update the test as well. The test is not applicable to DECIMALS.

Reply:
Thanks @karteekmurthys! I'll have to hold off on merging this PR until https://github.com/facebookincubator/velox/pull/2456 lands, we run with ASAN internally so the test introduced here is currently failing.